### PR TITLE
Bugfix/added a scroll bar to feedback editor

### PR DESCRIPTION
**Pull request by tessaRuckstuhl**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de. (Tested it on localhost)
- [x] Client: I added multiple screenshots/screencasts of my UI changes
### Motivation and Context
This change solves #1397 by expanding the text area to the size of its content.
### Description

I added the function `textareaAutogrow()` in `ngAfterViewInit()` for expanding the feedback textarea to the fitting size.
### Steps for Testing

Precondition: You have created a text exercise, which is ready for assessment
1. Log in to Artemis
2. Navigate to Course Management
3. Write multiple lines of text in it
4. Save the assessment and go back to the dashboard
5. Reopen the assessment
6. The textbox for feedback should still show all lines of feedback

### Screenshots
After reopening an assesment with use of multiple lines of feedback, it should look like this.
image

### CODE REVIEW

**Comment by filip-gregurevic** [comment]
I think that a better solution might be that the text area grows to a certain hight (something like 50px) to show more rows and rows that still don't fit in there are then shown on scroll. The scroll on a single line looks kinda weird

**Comment by jpbernius** [comment]
Maybe also try calling [`textareaAutogrow()`](https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.ts#L50) in [`textareaAutogrow()`](https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.ts#L43).

**Comment by tessaRuckstuhl** [comment]
Thanks for all your remarks on this! @jpbernius I tried this, but as the row is set in the textarea tag, the scrollheight property is set to the value of one row when calling `textareaAutogrow()` in `ngAfterViewInit()`. Not quite sure how to work around this. Creating a span element in `ngAfterViewInit()` and calculating the actual text height might work, but seems like an overcomplicated approach for solving this issue. Any ideas here?

**Comment by jpbernius** [comment]
Maybe calling this from a `setTimeout` in `ngAfterViewInit`:

```ts
setTimeout(() => this. textareaAutogrow());
```

**Comment by tessaRuckstuhl** [comment]
That works, thanks a lot!

File: .gitignore [hunk 1]
```diff
@@ -1,173 +0,0 @@
-######################
```
Comment (sleiss) [inline .gitignore:1]: Please restore this file.

File: src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.ts [hunk 2]
```diff
@@ -42,6 +42,16 @@ export class TextblockFeedbackEditorComponent implements AfterViewInit {
...
+        setTimeout(() => this.textareaOnloadSize());
+    }
+
+    /**
+     * Increase size of text area on page load to a max of 55px, depending on content
+     */
+    textareaOnloadSize(): void {
+        this.textareaElement.style.height = '0px';
+        this.textareaElement.style.height = `${this.textareaElement.scrollHeight}px`;
+        this.textareaElement.style.height = this.textareaElement.scrollHeight > 55 ? '55px' : `${this.textareaElement.scrollHeight}px`;
```
Comment (jpbernius) [inline src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.ts:54]: Is this the desired behavior?
Wouldn't it be preferred to have all elements at the needed height and not require scrolling here?

Comment (tessaRuckstuhl) [reply src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.ts:54]: I used @filip-gregurevic suggestion (see the comment above)j, to let it expand to a certain height and if less text is entered (<2 lines), then only this (1 line e.g.) is shown. Would you prefer a different behaviour?

Comment (jpbernius) [reply src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.ts:53]: So you are using this more than I am, so I am open to everything. :)
My personal preference is to reduce scrolling within scrollable content as much as possible.

Comment (tessaRuckstuhl) [reply src/main/webapp/app/exercises/text/assess-new/textblock-feedback-editor/textblock-feedback-editor.component.ts:53]: That's true.
As the text area expands (and stays expanded) while editing and only reduces in size on reopening, scrolling is reduced anyways. Though when reopening and decreasing the text box sizes, lets everyting appear very structured and clear look. So I think this is a nice compromise :).

**Comment by jpbernius** [comment]
image
In Safari on macOS, you do not see a scrollbar. This does make it unclear that there is more feedback. Either show the scrollbar in this case (It is possible to show it with CSS), or show the entire feedback.

**Comment by tessaRuckstuhl** [comment]
Ok, that's a shame. Just looked into the scrollbar styling, but this looks pretty weird when having the text area expanded and thus having a really long scrollbar (+firefox and internet explorer again support a different kind of css selector). Will update it, by showing the entire feedback as originally suggested by you :).

**Pull request merged**